implementation/elasticsearch/src/main/java/net/opentsdb/meta/NamespacedAggregatedDocumentSchema.java

```diff
@@ -229,13 +231,17 @@ protected static int countTagValueFilters(final QueryFilter filter, int count) {
       final TimeSeriesDataSourceConfig timeSeriesDataSourceConfig,
       final Span span) {
 
-    if (skip_meta.contains(timeSeriesDataSourceConfig.getNamespace()) &&
-    (System.currentTimeMillis()/1000L) - queryPipelineContext.query().startTime().epoch() >  META_RETENTION) {
-      NamespacedAggregatedDocumentResult result = new NamespacedAggregatedDocumentResult(
-          MetaResult.NO_DATA_FALLBACK, new Exception("Skipping meta store"), null);
-      return Deferred.fromResult(result);
-    }
+    Map<String, String> namespaces = tsdb.getConfig().getTyped(SKIP_META, NAMESPACE_FILTERS);
 
+    if (namespaces.containsKey(timeSeriesDataSourceConfig.getNamespace())) {
+      String retention = namespaces.get(timeSeriesDataSourceConfig.getNamespace());
+      long retention_ms = DateTime.parseDuration(retention);
+      if (System.currentTimeMillis() - queryPipelineContext.query().startTime().msEpoch() > retention_ms) {
+        NamespacedAggregatedDocumentResult result = new NamespacedAggregatedDocumentResult(
+            MetaResult.NO_DATA_FALLBACK, new Exception("Skipping meta store"), null);
```

Comment:
missed this, make a static runtime exception and return that to avoid the stack trace overhead.